### PR TITLE
Partial rewrite of our CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by bramceulemans**
### What is the reason for this PR?

We were missing some guidelines in our CONTRIBUTING.md

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -2,20 +2,32 @@
 
 If you've written a new formatter, added a new locale, or fixed a bug, your contribution is welcome!
 
-Before proposing a pull request, check the following:
+Before proposing a pull request, please these contribution guidelines.
```
Comment (pimjansen) [inline .github/CONTRIBUTING.md:5]:
```suggestion
Before proposing a pull request, please read these contribution guidelines.
```

**Review by pimjansen — CHANGES_REQUESTED** [review]

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (bramceulemans) [reply .github/CONTRIBUTING.md]: Fixed.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -2,20 +2,32 @@
...
+* Your code should follow [our coding standard](/.php_cs.dist). Run `make cs` to fix code style issues.
+* Your code should be covered by unit tests.
+* All checks must pass before your code is merged. Run `make test` to run unit tests.
+
+## Best Practices & Conventions
+
+When writing code for Faker, you should know about these conventions / best practices:
+
+* Never use `rand()` in your provider. Faker utilizes the Mersenne Twister Randomizer, so use `mt_rand()` or any of the base generators (`randomNumber`, `randomElement`, etc.) instead.
+* If you add a new provider which it embeds a lot of data (e.g. first names in a new language), please add a `@see` PHPDoc to the reference you used for this list (example [in the `en_US` locale](https://github.com/FakerPHP/Faker/blob/main/src/Faker/Provider/en_US/Text.php#L35)). This will ease future updates of the list and debates about the most relevant data for this provider.
```
Comment (krsriq) [inline .github/CONTRIBUTING.md:20]: If you add a new provider which ~~it~~ embeds

**Review by krsriq — CHANGES_REQUESTED** [review]

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (bramceulemans) [reply .github/CONTRIBUTING.md]: Fixed.

**Review by pimjansen — APPROVED** [review]

**Pull request merged**